packages/runtime/src/templating/view.ts

```diff
@@ -95,3 +99,160 @@ export class ViewFactory<T extends INode = INode> implements IViewFactory<T> {
     }
   }
 }
+
+type HasAssociatedViews = {
+  $views: ITemplateDefinition[];
+};
+
+export function view(v: ITemplateDefinition) {
+  return function<T extends Constructable>(target: T & Partial<HasAssociatedViews>) {
+    const views = target.$views || (target.$views = []);
+    views.push(v);
+  };
+}
+
+function hasAssociatedViews<T>(object: T): object is T & HasAssociatedViews {
+  return object && '$views' in object;
+}
+
+export const IViewLocator = DI.createInterface<IViewLocator>('IViewLocator')
+  .withDefault(x => x.singleton(ViewLocator));
+
+export interface IViewLocator {
+  getViewComponentForObject(object: ComposableObject | null | undefined, viewNameOrSelector?: string | ViewSelector): Constructable | null;
```

Comment:
```suggestion
  getViewComponentForObject<T extends ComposableObject>(
    object: T | null | undefined,
    viewNameOrSelector?: string | ViewSelector,
  ): ConstructableClass<T> | null;
```

This would retain the type information on the `prototype` and `constructor` (of the instance) properties of the type the user passed in.